openeo/extra/job_management/_thread_worker.py

```diff
@@ -207,8 +249,91 @@ def process_futures(self, timeout: Union[float, None] = 0) -> Tuple[List[_TaskRe
 
         self._future_task_pairs = to_keep
         return results, len(to_keep)
+    
+    def num_pending_tasks(self) -> int:
```

Comment:
abbreviating "number" to "num" is a bit off brand with the rest of the openEO client API.
In this age of large monitors and AI powered autocomplete there is little reason anymore to save on keystrokens at cost of readability.
I'd go for `pending_task_count` which is is just 2 characters more 😄